deploy/crds/monitor-app.yaml

```diff
@@ -1,41 +1,81 @@
-apiVersion: siddhi.io/v1alpha1
+# Change apiVersion v1alpha1 to v1alpha2 since we are changing existing entries
+apiVersion: siddhi.io/v1alpha2
 kind: SiddhiProcess
-metadata:
-  name: monitor-app
-  version: 1.0.0
-spec:
-  pod:
-   image: siddhiio/siddhi-runner-alpine
-   imageTag: 0.1.0
-  tls:
-    ingressSecret: siddhi-tls
-  env: 
-    - name: RECEIVER_URL
-      value: "http://0.0.0.0:8280/example"
-    - name: BASIC_AUTH_ENABLED
-      value: "false"
-  # apps:
-  #   - app1
-  #   - app2
-  query: |
-    @App:name("MonitorApp")
-    @App:description("Description of the plan") 
-    
-    @sink(type='log', prefix='LOGGER')
-    @source(type='http', receiver.url='${RECEIVER_URL}', basic.auth.enabled='${BASIC_AUTH_ENABLED}', @map(type='json'))
-    define stream DevicePowerStream (type string, deviceID string, power int);
-    
-    define stream MonitorDevicesPowerStream(deviceID string, power int);
-    @info(name='monitored-filter')
-    from DevicePowerStream[type == 'monitored']
-    select deviceID, power
-    insert into MonitorDevicesPowerStream;
-  siddhi.runner.configs: |
+metadata: 
+  name: nats-app
+spec: 
+
+  # apps entry can have siddhi apps as configMaps or direct scripts
+  apps: 
+    # - configMap: app
+    - script: |
+        @App:name("MonitorApp")
+        @App:description("Description of the plan") 
+  
+        @source(
+            type='http',
+            receiver.url='http://0.0.0.0:8080/example',
+            basic.auth.enabled='false',
+            @map(type='json')
+        )
+        define stream DevicePowerStream (type string, deviceID string, power int);
+        
+        @sink(type='log', prefix='LOGGER')
+        define stream MonitorDevicesPowerStream(sumPower long);
+        @info(name='monitored-filter')
+        from DevicePowerStream#window.time(100 min)
+        select sum(power) as sumPower
+        insert all events into MonitorDevicesPowerStream;
+  
+  # Replace previous pod spec with container. Now previous env entry also used under this container entry.
+  # Container entry can have siddhi runner image and envs.
+  container: 
+    env: 
+      - 
+        name: RECEIVER_URL
+        value: "http://0.0.0.0:8080/example"
+      - 
+        name: BASIC_AUTH_ENABLED
+        value: "false"
+      - 
+        name: NATS_URL
+        value: "nats://siddhi-nats:4222"
+      - 
+        name: NATS_DEST
+        value: siddhi
+      - 
+        name: NATS_CLUSTER_ID
+        value: siddhi-stan
+    image: "buddhiwathsala/siddhi-runner:0.1.1"
+
+  # imagePullSecret have the secret which can be used to download custom siddhi runner private docker images or 
```

Comment:
Shall we rephrase this as, 
Kubernetes [secret](https://kubernetes.io/docs/concepts/configuration/secret/) to pull Siddhi Runner image from private registries.